### ISSUE TITLE
No Roll?

### ISSUE THREAD

**Issue by Chillee** [ImperialRobotics/BoschIMU#1]
I notice that roll is set to 0 every time it's updated. Is there any particular reason for that?

**Comment by agilkes** [comment]
I wrote this code specifically to support the designs of FTC robots competing in this year's FIRST RES-Q competition. Yaw and pitch are critical for that purpose. A rolling FTC robot may well be in an unsafe orientation. I also wanted to get useful code into the hands of the FTC community as soon as possible. So the calculation and reporting of roll angles were left out.If you wish to add roll to the existing code, the technical literature references listed in the Java comments will help you.Regards,Alan   On 02/17/16, Chilleecom> wrote: I notice that roll is set to 0 every time it's updated. Is there any particular reason for that?—Reply to this email directly or view it on GitHub.

**Comment by Chillee** [comment]
Hey,Yep I agree. Roll is almost never useful in this competition (well maybe for determining how straight you are on the mountain). However, when we designed our robot, we put the sensor perpendicular to the robot's orientation. So "roll" is our pitch and "pitch" is our roll. I actually fixed it very quickly. Your code did 90% of the work in getting the pitch, I just needed to change a line or 2.
Thanks a lot for your code, by the way. I actually doubt I would have been able to get this IMU working without it, and the existence of your library is a large part of the reason our team decided to go with the Bosch IMU.
Thanks,Horace

```
  From: Alan M. Gilkes <notifications@github.com>
```

 To: ImperialRobotics/BoschIMU BoschIMU@noreply.github.com 
Cc: Chillee horacehe2007@yahoo.com
 Sent: Thursday, February 18, 2016 8:53 AM
 Subject: Re: [BoschIMU] No Roll? (#1)

 I wrote this code specifically to support the designs of FTC robots competing in this year's FIRST RES-Q competition. Yaw and pitch are critical for that purpose. A rolling FTC robot may well be in an unsafe orientation. I also wanted to get useful code into the hands of the FTC community as soon as possible. So the calculation and reporting of roll angles were left out.If you wish to add roll to the existing code, the technical literature references listed in the Java comments will help you.Regards,Alan   On 02/17/16, Chilleecom> wrote: I notice that roll is set to 0 every time it's updated. Is there any particular reason for that?—Reply to this email directly or view it on GitHub.—
